### PR TITLE
Feature/gdpr mass update and search opt in

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

These changes adds the ability to mass update the confirmed opt in field of the  primary email addressess for accounts, contacts, leads, and targets (prospects).

I have also added the ability to search in the list view by the confirm opt in field. In order to achieve this i have had to change the default value from empty to "not-opt-in". This is due to the way the filter skips setting up a where clause when the value is empty. This means that we will need to ensure that the existing behaviour stays in tack.

In order to get the test to work, I have merged the RC2 changes from the develop branch in this pull request. My changes are:
- Updating code to SugarEmailAddress::COI_STAT_DISABLED
- Reformatting searchdefs for accounts, contacts, leads and prospects
- Adding opt in search to prospects
- Adding opt in search to leads
- Adding opt in search to contacts
- Adding opt in search to accounts
- Updating comments in ViewList
- Changing default value for confirm_opt_in
- Switched opt in check box with a select on the mass update form
- Adding opt in primary email checkbox to mass update form
- Adding LBL_OPT_IN_FLAG_PRIMARY

please only review my changes.

## Motivation and Context

- GDPR

## How To Test This

* Install SuiteCRM (since there are changes in the email address table)
* Enable opt in settings in the email settings
* got to an accounts/contacts/leads/targets(prospects)
* select some records
* bulk actions -> mass update
* set opt in primary address to yes
* select update
* filter
* advanced
* select opt in and/or confirmed opt in
* select search

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by Dillon-Brown** [comment]
Assessed

File: include/MassUpdate.php [hunk 1]
```diff
@@ -308,25 +308,41 @@ function handleMassUpdate(){
 							$newbean->parent_type = null;
 						}
 
-						$email_address_id = '';
+						// get primary address id
+                        $email_address_id = '';
+						$email_address_value = array();
```
Comment (PedroErnst) [inline include/MassUpdate.php]: Tabs

File: include/MassUpdate.php [hunk 2]
```diff
@@ -350,9 +366,20 @@ function handleMassUpdate(){
 
 						$newbean->save($check_notify);
 						if (!empty($email_address_id)) {
-	    					$query = "UPDATE email_addresses SET opt_out = {$optout_flag_value} where id = '{$emailAddressRow['email_address_id']}'";
-	    					$GLOBALS['db']->query($query);
-
+						    /** @var EmailAddress $primaryEmailAddress */
```
Comment (PedroErnst) [inline include/MassUpdate.php]: Tabs

File: include/MassUpdate.php [hunk 3]
```diff
@@ -350,9 +366,20 @@ function handleMassUpdate(){
...
+                            $primaryEmailAddress = BeanFactory::getBean('EmailAddresses', $email_address_id);
+						    if ($setOptOutPrimaryEmailAddress) {
+                                $primaryEmailAddress->opt_out = (bool)$optout_flag_value;
+                            }
+
+                            if ($setOptInPrimaryEmailAddress) {
+						        if($optin_flag_value === true) {
```
Comment (PedroErnst) [inline include/MassUpdate.php]: Tabs

**Comment by daniel-samson** [comment]
The css files are intentional, as they were merged in from the develop branch. I have not modified them.

File: include/MassUpdate.php (hunk 1 above)
Comment (daniel-samson) [reply include/MassUpdate.php]: i have reformatted this file

File: include/MassUpdate.php (hunk 2 above)
Comment (daniel-samson) [reply include/MassUpdate.php]: i have reformatted this file

File: include/MassUpdate.php (hunk 3 above)
Comment (daniel-samson) [reply include/MassUpdate.php]: i have reformatted this file

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**